test/tstQueryTreeCallbacks.cpp

```diff
@@ -75,54 +75,40 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(callback_spatial_predicate, TreeTypeTraits,
 
   int const n = 10;
-  Kokkos::View<ArborX::Point *, DeviceType> points("points", n);
-  Kokkos::View<Kokkos::pair<int, float> *, DeviceType> ref("ref", n);
-  ArborX::Point const origin = {{0., 0., 0.}};
-  Kokkos::parallel_for(
-      Kokkos::RangePolicy<ExecutionSpace>(0, n), KOKKOS_LAMBDA(int i) {
-        points(i) = {{(double)i, (double)i, (double)i}};
-        ref(i) = {i, (float)ArborX::Details::distance(points(i), origin)};
-      });
-
-  Tree const tree(ExecutionSpace{}, points);
-  {
-    Kokkos::View<Kokkos::pair<int, float> *, DeviceType> custom("custom", 0);
-    Kokkos::View<int *, DeviceType> offset("offset", 0);
-    ArborX::query(tree, ExecutionSpace{},
-                  makeIntersectsBoxQueries<DeviceType>({
-                      static_cast<ArborX::Box>(tree.bounds()),
-                  }),
-                  CustomInlineCallback<DeviceType>{points}, custom, offset);
-
-    auto custom_host =
-        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, custom);
-    auto ref_host =
-        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, ref);
-    auto offset_host =
-        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offset);
-    BOOST_TEST(make_compressed_storage(offset_host, custom_host) ==
-                   make_compressed_storage(offset_host, ref_host),
-               tt::per_element());
-  }
+  ArborX::Point const origin{0, 0, 0};
+  std::vector<ArborX::Point> pts;
+  std::vector<Kokkos::pair<int, float>> values;
+  for (int i = 0; i < 10; ++i)
   {
-    Kokkos::View<Kokkos::pair<int, float> *, DeviceType> custom("custom", 0);
-    Kokkos::View<int *, DeviceType> offset("offset", 0);
-    ArborX::query(tree, ExecutionSpace{},
-                  makeIntersectsBoxQueries<DeviceType>({
-                      static_cast<ArborX::Box>(tree.bounds()),
-                  }),
-                  CustomPostCallback<DeviceType>{points}, custom, offset);
-
-    auto custom_host =
-        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, custom);
-    auto ref_host =
-        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, ref);
-    auto offset_host =
-        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, offset);
-    BOOST_TEST(make_compressed_storage(offset_host, custom_host) ==
-                   make_compressed_storage(offset_host, ref_host),
-               tt::per_element());
+    ArborX::Point pt{{(double)i, (double)i, (double)i}};
+    values.emplace_back(i, ArborX::Details::distance(pt, origin));
+    pts.emplace_back(std::move(pt));
   }
+  std::vector<int> offsets = {0, n};
+
+  Kokkos::View<ArborX::Point *, DeviceType> points(
+      Kokkos::view_alloc("Test::points", Kokkos::WithoutInitializing),
+      pts.size());
+  Kokkos::deep_copy(points,
+                    Kokkos::View<ArborX::Point const *, Kokkos::HostSpace,
+                                 Kokkos::MemoryTraits<Kokkos::Unmanaged>>{
+                        pts.data(), pts.size()});
```

Comment:
Not a fan of this change. Would very much prefer:
```c++
  Kokkos::View<ArborX::Point *, DeviceType> points(Kokkos::ViewAllocateWithoutInitializing("points"), n);
  ArborX::Point const origin = {{0., 0., 0.}};
  Kokkos::parallel_for(
      Kokkos::RangePolicy<ExecutionSpace>(0, n), KOKKOS_LAMBDA(int i) {
        points(i) = {{(double)i, (double)i, (double)i}};
      });
   auto points_host = Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, points);

   std::vector<Kokkos::pair<int, float>> ref;
   for (int i = 0; i < n; ++i)
      ref.emplace_back(i, ArborX::Details::distance(points_host(i), origin));
   std::vector<int> offset = {0, n};
```
This comment is applicable to all other functions.